### ISSUE TITLE
Loss is negative? Reproducing MapNet on RobotCar `loop` dataset

### ISSUE THREAD

**Issue by Jiankai-Sun** [NVlabs/geomapnet#10]
1. After setting up the environment as this repo describes, I train MapNet on `loop` dataset from scratch by running
```python
python train.py --dataset RobotCar --scene loop --config_file configs/mapnet.ini --model mapnet --device 0 --learn_beta --learn_gamma
```
After that, I get the model `logs/RobotCar_loop_mapnet_mapnet_learn_beta_learn_gamma/epoch_300.pth.tar ` and evaluate the model on `loop` but the result seems not as good as the paper reported. Do you have any suggestions?
```bash
$ python eval.py --dataset RobotCar --scene loop --config_file configs/mapnet.ini --model mapnet --weights logs/RobotCar_loop_mapnet_mapnet_learn_beta_learn_gamma/epoch_300.pth.tar --val --output_dir ../results/

../dataset_loaders/robotcar_sdk/interpolate_poses.py:143: RuntimeWarning: divide by zero encountered in divide
  (pose_timestamps[upper_indices] - pose_timestamps[lower_indices])
Loaded weights from logs/RobotCar_loop_mapnet_mapnet_learn_beta_learn_gamma/epoch_300.pth.tar
Running mapnet on VAL data
../common/train.py:339: UserWarning: volatile was removed and now has no effect. Use `with torch.no_grad():` instead.
  data_var = Variable(data, volatile=(not train), requires_grad=train)
Image 0 / 3356
Image 200 / 3356
Image 400 / 3356
Image 600 / 3356
Image 800 / 3356
Image 1000 / 3356
Image 1200 / 3356
Image 1400 / 3356
Image 1600 / 3356
Image 1800 / 3356
Image 2000 / 3356
Image 2200 / 3356
Image 2400 / 3356
Image 2600 / 3356
Image 2800 / 3356
Image 3000 / 3356
Image 3200 / 3356
Error in translation: median 150.43 m,  mean 142.86 m
Error in rotation: median 57.09 degrees, mean 77.71 degree
```

2. BTW, a strange phenomenon is that the training loss is negative. As the attachment shows: [log.txt](https://github.com/NVlabs/geomapnet/files/2735585/log.txt)

**Comment by samarth-robo** [comment]
Hi @Jiankai-Sun, thanks for raising this issue! I believe I have found the issue (a change in the code that happened while cleaning it up for release). I will push a commit in a couple of days, once I've re-trained a model for verification.

**Comment by samarth-robo** [comment]
This is fixed in commit f7b912a5b1df3ea2196cb1f62b901dedd5828081.
```
$ python eval.py --dataset RobotCar --scene loop --model mapnet --config configs/mapnet.ini --device 0 --weights logs/RobotCar_loop_mapnet_mapnet_learn_beta_learn_gamma/epoch_300.pth.tar --val
Loaded weights from logs/RobotCar_loop_mapnet_mapnet_learn_beta_learn_gamma/epoch_300.pth.tar
Running mapnet on VAL data
../dataset_loaders/robotcar_sdk/interpolate_poses.py:142: RuntimeWarning: divide by zero encountered in divide
  (pose_timestamps[upper_indices] - pose_timestamps[lower_indices])
Image 0 / 3356
Image 200 / 3356
Image 400 / 3356
Image 600 / 3356
Image 800 / 3356
Image 1000 / 3356
Image 1200 / 3356
Image 1400 / 3356
Image 1600 / 3356
Image 1800 / 3356
Image 2000 / 3356
Image 2200 / 3356
Image 2400 / 3356
Image 2600 / 3356
Image 2800 / 3356
Image 3000 / 3356
Image 3200 / 3356
Error in translation: median 5.21 m,  mean 9.18 m
Error in rotation: median 1.43 degrees, mean 3.86 degree
```
